### PR TITLE
Add tutorial `13_trotterQRTE.ipynb`

### PR DESCRIPTION
**Pull request by ialsina**
### Summary

Qiskit Algorithms tutorial for Trotter QRTE.

### Details and comments

The original discussion can be found in [the original Qiskit PR](https://github.com/Qiskit/qiskit-tutorials/pull/1465).

### CODE REVIEW

**Review by woodsp-ibm — COMMENTED** [review]
CI is failing due to the fact that we spell check content that becomes part of the published docs and this ends up failing. `Colorplot` I mentioned above is a title I think we can alter to fix that and at the same time make it more informative as a title. Others like `exponentials` and the names should be added the .pylintdict in the root (lowercase and in alphabetic order so its easier for us to manage). The spelling failures are in the CI build details. Names we often end up doing this with - and sometimes with plurals `exponentials` - its an adjective which I guess is why the dictionary does not have a plural form - but I get that its used sometimes as you have it so adding it to the dict will allow it to stay like this ie `matrix exponentials`

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (woodsp-ibm) [inline docs/tutorials/13_trotterQRTE.ipynb:52]:
```suggestion
    "First, we will create a function that takes in the system parameters $L$, $J$, $h$ and $\\alpha$, and returns our Hamiltonian as a `SparsePauliOp`. A [SparsePauliOp](https://qiskit.org/documentation/stubs/qiskit.quantum_info.SparsePauliOp.html) is a sparse representation of an operator in terms of weighted [Pauli](https://qiskit.org/documentation/stubs/qiskit.quantum_info.Pauli.html) terms."
```
Otherwise it formats badly
image

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (woodsp-ibm) [inline docs/tutorials/13_trotterQRTE.ipynb:501]:
```suggestion
    "on each one of the timesteps used by Trotter. We compute this exponential using SciPy's [linalg.expm](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.expm.html) function, and then we let the initial system evolve using the `Statevector`'s [evolve](https://qiskit.org/documentation/stubs/qiskit.quantum_info.Statevector.evolve.html) method. This becomes too costly to be performed on larger systems very quickly."
```
image

I also change the `scipy` to `SciPy` since the latter is how the project itself has it

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (woodsp-ibm) [inline docs/tutorials/13_trotterQRTE.ipynb:674]:
```suggestion
    "There exists a second-order formula, called the Suzuki-Trotter decomposition [3], and can be used in Qiskit by means of the [SuzukiTrotter](https://qiskit.org/documentation/stubs/qiskit.synthesis.SuzukiTrotter.html) class. Using this formula, a second order decomposition is:\n",
```

**Review by woodsp-ibm — COMMENTED** [review]
I think this is all the other link formatting issues like the one I first commented on - not sure if you saw the note I had put in there about others.

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (woodsp-ibm) [inline docs/tutorials/13_trotterQRTE.ipynb:674]:
```suggestion
    "There exists a second-order formula, called the Suzuki-Trotter decomposition [3], and can be used in Qiskit by means of the [SuzukiTrotter](https://qiskit.org/documentation/stubs/qiskit.synthesis.SuzukiTrotter.html) class. Using this formula, a second order decomposition is:\n",
```
I had marked this in the review above but it came out odd - hopefully this shows the change correctly now

Comment (woodsp-ibm) [inline docs/tutorials/13_trotterQRTE.ipynb:674]: I cannot seem to mark this as a change without it showing in pink text, what is being changed, one of the images here. What I was trying to do a suggestion for was from 
```
 [`SuzukiTrotter` class](https://qiskit.org/documentation/stubs/qiskit.synthesis.SuzukiTrotter.html).
```
to
```
 [SuzukiTrotter](https://qiskit.org/documentation/stubs/qiskit.synthesis.SuzukiTrotter.html) class.
```
in the complete line 674.

Comment (woodsp-ibm) [reply docs/tutorials/13_trotterQRTE.ipynb:674]: Even now when I put it on line 674 its showing up for line 631 here - no idea!

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (ialsina) [reply docs/tutorials/13_trotterQRTE.ipynb:191]: You are totally right about this cell output. Nothing ended up convincing me. There was already a comment on [the original discussion](https://github.com/Qiskit/qiskit-tutorials/pull/1465) that was never resolved. Let me paste part of it below:

> [...] I would draw the quantum circuit, using `'mpl'` as the output argument. However, this quantum circuit isn't composed of standard gates. One would need to decompose it two times, and then decompose it specifying which gates to decompose, via a line like `result.evolved_state.decompose(reps=2).decompose('disentangler_dg').decompose('multiplex1_reverse_dg').draw('mpl')`
but this seems a bit specific for a tutorial like this. For this reason, I decided to show that it is (an instance of) a QuantumCircuit. [...]

I attach below a couple of images that will hopefully give an impression of why I am not a big fan of this solution. Let me know what are your thoughts on this! :)

image
image

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (ialsina) [reply docs/tutorials/13_trotterQRTE.ipynb]: Perfect Steve, thank you for spotting them! That's strange about the Github diff. Good call though -- I mark the conversation resolved as commit `c75cf01` should address this.

File: .pylintdict
Comment (ElePT) [inline .pylintdict:186]: Missing one more name in the pylintdict:

```suggestion
magnus
marecek
masuo
```

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (ElePT) [reply docs/tutorials/13_trotterQRTE.ipynb:191]: I suggest drawing both the "raw" output of `evolved_state` and the decomposed version + a short explanation of why we are decomposing (to show the standard gates that are used in this circuit).

Comment (ialsina) [reply docs/tutorials/13_trotterQRTE.ipynb]: After a brief chat with Elena, for now we are going with both: `result.evolved_state.draw('mpl')` and the version that brings all gates to standard Circuit Library gates. That comes in commit `0d31c9f`. :+1:

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (ElePT) [inline docs/tutorials/13_trotterQRTE.ipynb:200]: Tiny tiny suggestion: 

```suggestion
    "By sequentially decomposing the circuit, we can show it in terms of the gates of the Qiskit [Circuit Library](https://qiskit.org/documentation/apidoc/circuit_library.html)."
```

File: docs/tutorials/13_trotterQRTE.ipynb
Comment (ElePT) [inline docs/tutorials/13_trotterQRTE.ipynb:200]:
```suggestion
    "By sequentially decomposing the circuit, we can show it in terms of Qiskit's [Circuit Library](https://qiskit.org/documentation/apidoc/circuit_library.html) standard gates."
```

**Review by ElePT — DISMISSED** [review]
LGTM! Thank you very much @ialsina for this great work. I will wait for @woodsp-ibm approval, but I think it's ready to be merged.

**Review by woodsp-ibm — APPROVED** [review]

**Comment by woodsp-ibm** [comment]
@ialsina LGTM - thanks for work and contributing this tutorial.

**Pull request merged**
